lib/puppet/provider/reboot/windows.rb

```diff
@@ -2,8 +2,8 @@
 require 'open3'
 
 Puppet::Type.type(:reboot).provide :windows do
-  confine :operatingsystem => :windows
-  defaultfor :operatingsystem => :windows
+  confine :kernel => :windows
+  defaultfor :kernel => :windows
```

Comment:
Incidentally, in doing that research, I discovered that you can do `confine :feature => :posix`, which might be a better option for the Linux provider than `confine :kernel => :linux`.  When doing research for the Linux provider, I looked into some other OS environments and determined that pretty much everything that's descended from UNIX appears to have the same syntax for `shutdown`.  The Linux provider could thus be trivially converted to a general POSIX provider so long as we make sure to note in the documentation that systems not explicitly listed as supported are not tested by Puppet Labs and should be tested by the user before deployment.  Thoughts?